exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterWorker.java

```diff
@@ -128,32 +151,49 @@ void export() {
         viewDataList.add(viewManager.getView(view.getName()));
       }
     }
+
     List<TimeSeries> timeSeriesList = Lists.newArrayList();
     for (ViewData viewData : viewDataList) {
       timeSeriesList.addAll(
           StackdriverExportUtils.createTimeSeriesList(viewData, monitoredResource));
     }
-
     for (List<TimeSeries> batchedTimeSeries :
         Lists.partition(timeSeriesList, MAX_BATCH_EXPORT_SIZE)) {
-      // Batch export 3 TimeSeries at one call, to avoid exceeding RPC header size limit.
-      CreateTimeSeriesRequest request =
-          CreateTimeSeriesRequest.newBuilder()
-              .setNameWithProjectName(projectName)
-              .addAllTimeSeries(batchedTimeSeries)
-              .build();
-      try {
+      Span span = tracer.getCurrentSpan();
+      span.addAnnotation("Export Stackdriver TimeSeries.");
+      try (Scope scope = tracer.withSpan(span)) {
+        // Batch export 3 TimeSeries at one call, to avoid exceeding RPC header size limit.
+        CreateTimeSeriesRequest request =
+            CreateTimeSeriesRequest.newBuilder()
+                .setNameWithProjectName(projectName)
+                .addAllTimeSeries(batchedTimeSeries)
+                .build();
         metricServiceClient.createTimeSeries(request);
+        span.addAnnotation("Finish exporting TimeSeries.");
+      } catch (ApiException e) {
+        logger.log(Level.WARNING, "ApiException thrown when exporting TimeSeries.", e);
+        span.addAnnotation("ApiException thrown when exporting TimeSeries.");
+        span.setStatus(Status.CanonicalCode.valueOf(e.getStatusCode().getCode().name()).toStatus());
       } catch (Throwable e) {
         logger.log(Level.WARNING, "Exception thrown when exporting TimeSeries.", e);
+        span.addAnnotation("Exception thrown when exporting TimeSeries.");
+        span.setStatus(Status.UNKNOWN);
       }
     }
+    tracer.getCurrentSpan().end();
```

Comment:
No need anymore. This is called in the main loop (by the scoped span or by the new code I proposed).

Reply:
Done.